server/resources.py

```diff
@@ -61,5 +61,43 @@ def post(self, lecture_id):
             'id': comment.id
         }
 
+
+class EngagementListResource(Resource):
+    def post(self, lecture_id):
+        lecture = Lecture.query.filter(Lecture.id == lecture_id).first()
+
+        if not lecture:
+            abort(404, message="Lecture {} does not exist".format(lecture_id))
+
+        parser = reqparse.RequestParser()
+        parser.add_argument('challenge', required=True, type=float)
+        parser.add_argument('interest', required=True, type=float)
+        parser.add_argument('time', required=True, type=str)
+        args = parser.parse_args()
+
+        challenge = args.challenge
+        interest = args.interest
+
+        if (challenge < 0.0 or challenge > 1.0):
```

Comment:
Oh, I have never seen that. Is it preferred? To me it easier to read how it currently is, but it might just be me not being used to the other way.